src/Tasks/Microsoft.NET.Build.Tasks/RunReadyToRunCompiler.cs

```diff
@@ -149,16 +220,25 @@ private string GetAssemblyReferencesCommands()
             return result.ToString();
         }
 
+        protected override string GenerateCommandLineCommands()
+        {
+            if (ActuallyUseCrossgen2 && !string.IsNullOrEmpty(DotNetHostPath))
+            {
+                return Crossgen2Tool.ItemSpec;
```

Comment:
I didn't want to introduce the `Quote` function, but don't we still need to quote this one?
```C#
    return $"\"{Crossgen2Tool.ItemSpec}\"";
```

Reply:
I guess I misunderstood you, I thought you were saying that the quoting is undesirable as it's done internally in the base class. I'll fix that.

Reply:
Fixed in 7th commit.